cli_helpers/tabular_output/preprocessors.py

```diff
@@ -83,6 +83,7 @@ def align_decimals(data, headers, column_types=(), **_):
 
     """
     pointpos = len(headers) * [0]
+    data = list(data)
```

Comment:
sadly no, the whole data iterator needs to be iterated to check the alignment before the first row can be outputted.